core/Tracker/RequestHandlerTrait.php

```diff
@@ -31,10 +31,12 @@ protected function checkSiteExists(Request $request): void
         try {
             $request->getIdSite();
         } catch (UnexpectedWebsiteFoundException $e) {
-            // we allow 0... the request will fail anyway as the site won't exist... allowing 0 will help us
-            // reporting this tracking problem as it is a common issue. Otherwise we would not be able to report
-            // this problem in tracking failures
-            StaticContainer::get(Failures::class)->logFailure(Failures::FAILURE_ID_INVALID_SITE, $request);
+            $idSite = $this->request->getRawParams()['idsite'] ?? null;
```

Comment:
```suggestion
            $idSite = $request->getRawParams()['idsite'] ?? null;
```

The request "should" always be set in the classes using the trait, but as we also get it as a function parameter we don't need rely on an instance property.